test_basic.py

```
@@ -0,0 +1,289 @@
+        
+        with patch('pipeline.extract_fields', return_value={}):
+            with tempfile.NamedTemporaryFile(suffix='.png', delete=False) as tmp:
+                test_path = Path(tmp.name)
+            
+            try:
+                # Mock sys.argv
+                original_argv = sys.argv
+                sys.argv = ['pipeline.py', str(test_path)]
+                
+                with patch('builtins.print') as mock_print:
+                    with patch('pipeline._validate_file_format'):
```

Ruff (0.11.9)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)